pkg/api/validation/validation.go

```diff
@@ -1024,6 +1024,38 @@ func validatePathNoBacksteps(targetPath string, fldPath *field.Path) field.Error
 	return allErrs
 }
 
+// validateMountPropagation verifies that MountPropagation field is valid and
+// allowed for given container.
+func validateMountPropagation(mountPropagation *api.MountPropagationMode, container *api.Container, fldPath *field.Path) field.ErrorList {
```

Comment:
The api changes guideline for alpha fields also suggestss clearing the field before persisting to the storage. https://github.com/kubernetes/community/blob/master/contributors/devel/api_changes.md#alpha-field-in-existing-api-version

Not sure if it's needed but it's worth checking out.

Reply:
I did not find any feature gate checks in `pkg/registry/core`. Is really required?

Reply:
there is discussion about clearing alpha fields in https://github.com/kubernetes/kubernetes/pull/50924. I'll wait for result and fix mountPropagation field when necessary.